taipy/core/scenario/scenario.py

```diff
@@ -638,40 +638,57 @@ def _remove_tag(self, tag: str) -> None:
     def _get_set_of_tasks(self) -> Set[Task]:
         return set(self.tasks.values())
 
-    def __get_data_nodes(self) -> Dict[str, DataNode]:
-        data_nodes_dict = self.__get_additional_data_nodes()
-        for _, task in self.__get_tasks().items():
+    def _get_existing_data_nodes(self, additional:bool=False) -> Dict[str, DataNode]:
+        if additional:
+            return self.__get_additional_data_nodes(False)
```

Comment:
Verbose for more readable code

```suggestion
            return self.__get_additional_data_nodes(raise_not_existing=False)
```